### PR TITLE
idsk : init at unstable-2018-02-11

### PR DESCRIPTION
**Pull request by bignaux**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/filesystems/idsk/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+
+{ stdenv, fetchFromGitHub, cmake }:
+
+stdenv.mkDerivation rec {
+
+	pname = "idsk";
+  version = "0.16";
+  rev = "1846729ac3432aa8c2c0525be45cfff8a513e007";
+  short_rev = "${builtins.substring 0 7 rev}";
+	name = "${pname}-${version}-${short_rev}";
```
Comment (lheckemann) [inline pkgs/tools/filesystems/idsk/default.nix]: indentation is off here

**Comment by lheckemann** [comment]
@GrahamcOfBorg build idsk

File: pkgs/tools/filesystems/idsk/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+
+{ stdenv, fetchFromGitHub, cmake }:
+
+stdenv.mkDerivation rec {
+
+	pname = "idsk";
+  version = "0.16";
```
Comment (nlewo) [inline pkgs/tools/filesystems/idsk/default.nix]: Where did you find the version number? I'm able to get it from there github.
When then revision is not a release, we have to use `version = "unstable-YYYY-MM-DD" as mentionned there https://nixos.org/nixpkgs/manual/#sec-package-naming

Comment (bignaux) [reply pkgs/tools/filesystems/idsk/default.nix]: It refers to the version display : 
"iDSK version 0.16 (by Demoniak, Sid, PulkoMandy), http://github.com/cpcsdk"
But i will change to unstable-YYYY-MM-DD if it more compliant.

**Comment by bignaux** [comment]
I've fixed version, but i'm unhappy with the idsk-unstable-2018-02-11 name. giving date show it's not a release, unstable is redundant and give no information about what it is.

**Comment by lheckemann** [comment]
https://nixos.org/nixpkgs/manual/#sec-syntax

> - Use 2 spaces of indentation per indentation level in Nix expressions, 4 spaces in shell scripts.
> - Do not use tab characters, i.e. configure your editor to use soft tabs. For instance, use `(setq-default indent-tabs-mode nil)` in Emacs. Everybody has different tab settings so it’s asking for trouble.

I believe the -unstable naming convention is to make sure that in the presence of a stable and an unstable version of a package, nix-env will not automatically select the unstable one because it has a higher version number. See the [description](https://nixos.org/nix/manual/#description-1); this isn't a great behaviour IMO and nobody who knows the difference will prefer `nix-env -i` over `nix-env -iA` but hey, such is life. I also agree that if upstream has some sort of version number it should be included. I don't know the exact policies around this though.

**Comment by nlewo** [comment]
I also don't know if there is a policy regarding this case. But I think we can consider this is a release. When we use a tag to fetch a release, this tag is mutable and set by application developers. This is really close from this displayed version number.
@bignaux could you switch back to the version number (instead of `unstable-date`)?

**Comment by bignaux** [comment]
I'd prefer keep the name as it, and merge, and later i found time to fix bug in idsk and ask for a bug release.

**Comment by nlewo** [comment]
Thanks

**Pull request merged**